File: tools-ui/src/app/model/crl-equipment.ts
export class CrlEquipment {
  id: number;
  name: string;
  description: string;
  serialNumber: string;
  borrower: string;

  isHidden = false;


	constructor(id: number, name: string, description: string, serialNumber: string, borrower: string) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.serialNumber = serialNumber;
    this.borrower = borrower;
  }
    
  static empty() {
    return new CrlEquipment(0, '', '', '', '');
  }

  public trimAll() {
    this.name = this.name.trim();
    this.description = this.description.trim();
    this.serialNumber = this.serialNumber.trim();
  }
}

File: tools-server/src/main/java/com/fmning/tools/domain/CrlEquipment.java
package com.fmning.tools.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;
import java.time.Instant;

@Entity
@Table(name="tl_crl_equipments")
@DynamicUpdate
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CrlEquipment {

    @Id
    @Column(name="id")
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private int id;

    @Column(name="name")
    private String name;

    @Column(name="description")
    private String description;

    @Column(name="picture")
    private String picture;

    @Column(name="serial_number")
    private String serialNumber;

    @Column(name="created")
    private Instant created;

    @Column(name="borrower")
    private String borrower;
}
